### PR TITLE
[CI][Jenkins] Remove OpenCL CI Stages

### PR DESCRIPTION
**Pull request by JehandadKhan**

### CODE REVIEW

**Comment by JehandadKhan** [comment]
FYI @atamazov

**Comment by atamazov** [comment]
@JehandadKhan @junliume Prior reviewing -- 
- Do we need to perform nonzero amount of work to support OCL BE in the code?
- Have we decided to drop support for OCL BE permanently? 
- What we can drop right away is tuning for OCL BE (system databases and precompiled binaries), I believe.

**Review by junliume — APPROVED** [review]

**Review by junliume — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -615,45 +589,6 @@ pipeline {
                 expression { params.BUILD_SMOKE_FP32 && params.DATATYPE_FP32 }
             }
             parallel{
-               stage('Fp32 OpenCL Debug + Codecov AnyGPU') {
```
Comment (atamazov) [inline Jenkinsfile:618]: Side effect: no CODECOV tests anymore.

File: Jenkinsfile [hunk 2]
```diff
@@ -680,19 +615,6 @@ pipeline {
                         buildHipClangJobAndReboot(build_type: 'debug', config_targets: Smoke_targets)
                     }
                 }
-                stage('Fp32 OpenCL Debug gfx1030') {
```
Comment (atamazov) [inline Jenkinsfile:683]: We do not have FP32 gfx1030 Smoke test anymore. This must be converted to HIP BE instead of removing.

File: Jenkinsfile [hunk 3]
```diff
@@ -826,32 +748,6 @@ pipeline {
                 expression { params.BUILD_SMOKE_FP16_BF16_INT8 }
             }
             parallel{
-                stage('Fp16 OpenCL Vega20') {
```
Comment (atamazov) [inline Jenkinsfile:829]: We do not have FP16 Vega20 Smoke test anymore. This must be converted to HIP BE instead of removing.

File: Jenkinsfile [hunk 4]
```diff
@@ -826,32 +748,6 @@ pipeline {
...
-                    }
-                    options {
-                        retry(2)
-                    }
-                    agent{ label rocmnode("vega20") }
-                    steps{
-                        buildHipClangJobAndReboot(compiler: 'g++', setup_flags: Fp16_flags, config_targets: Smoke_targets)
-                    }
-                }
-                stage('Int8 OpenCL Vega20') {
```
Comment (atamazov) [inline Jenkinsfile:842]: We do not have INT8 Vega20 Smoke test anymore. This must be converted to HIP BE instead of removing.

File: Jenkinsfile [hunk 5]
```diff
@@ -1097,68 +993,6 @@ pipeline {
...
-                    options {
-                        retry(2)
-                    }
-                    agent{ label rocmnode("gfx90a") }
-                    steps{
-                        buildHipClangJobAndReboot(compiler: 'g++', setup_flags: Full_test, build_install: "true")
-                    }
-                }
-            }
-        }
```
Comment (atamazov) [inline Jenkinsfile:1161]: Note: INT8 Full tests are missing for gfx908, gfx90a, gfx1030.

**Review by atamazov — COMMENTED** [review]
@JehandadKhan @junliume We removed some tests that should have been moved to HIP BE. Also I see some lack of INT8 tests.

**Pull request merged**